### ISSUE TITLE
FileUpload: Unsupported Content-Type [application/x-www-form-urlencoded; charset=UTF-8], expected [multipart/form-data]

### ISSUE THREAD

**Issue by martin654** [primefaces/primefaces#12021]
### Describe the bug

Hi, 
I am trying to upgrade to PF 14.0.0 from 13.0.9 and I found a problem with p:fileUpload. I use component like this:
```
        <h:form id="frmTest">
          <div class="grid">
            <div class="col-12 lg:col-6">


              <p:fileUpload mode="advanced" multiple="true" auto="true" process="@this" update="@form"
                            label="Attachments" listener="#{fileUploadView.handleFileUpload}">
              </p:fileUpload>

              <p:commandButton ajax="true" process="@form" update="@this" value="process form" />
            </div>
          </div>
        </h:form>
```
The commandButton is supposed to perform some other action on page (not related with fileUpload), but it throws exception when submitted:
```
Caused by: javax.servlet.ServletException: Unsupported Content-Type [application/x-www-form-urlencoded; charset=UTF-8], expected [multipart/form-data]
	at org.eclipse.jetty.server.Request.getParts(Request.java:2419)
	at org.eclipse.jetty.server.Request.getPart(Request.java:2410)
	at org.primefaces.component.fileupload.NativeFileUploadDecoder.createUploadedFile(NativeFileUploadDecoder.java:64)
	at org.primefaces.component.fileupload.AbstractFileUploadDecoder.decodeAdvanced(AbstractFileUploadDecoder.java:86)
	at org.primefaces.component.fileupload.AbstractFileUploadDecoder.decode(AbstractFileUploadDecoder.java:54)
	... 60 more

kvě 29, 2024 10:14:02 DOP. com.sun.faces.context.AjaxExceptionHandlerImpl handlePartialResponseError
SEVERE: javax.servlet.ServletException: Unsupported Content-Type [application/x-www-form-urlencoded; charset=UTF-8], expected [multipart/form-data]
	at org.eclipse.jetty.server.Request.getParts(Request.java:2419)
	at org.eclipse.jetty.server.Request.getPart(Request.java:2410)
	at org.primefaces.component.fileupload.NativeFileUploadDecoder.createUploadedFile(NativeFileUploadDecoder.java:64)
	at org.primefaces.component.fileupload.AbstractFileUploadDecoder.decodeAdvanced(AbstractFileUploadDecoder.java:86)
	at org.primefaces.component.fileupload.AbstractFileUploadDecoder.decode(AbstractFileUploadDecoder.java:54)
	at org.primefaces.component.fileupload.FileUploadRenderer.decode(FileUploadRenderer.java:58)
	at javax.faces.component.UIComponentBase.decode(UIComponentBase.java:514)
	at javax.faces.component.UIInput.decode(UIInput.java:798)
```

This worked fine in PF 13.0.9

### Reproducer
Attached reproducer:
[fileupload.zip](https://github.com/primefaces/primefaces/files/15481114/fileupload.zip)

### Expected behavior

The same as in 13.0.9

### PrimeFaces edition

Elite

### PrimeFaces version

14.0.0

### JSF implementation

Mojarra

### JSF version

4.0

### Java version

17

**Comment by tandraschko** [comment]
Did you test from source or checking issues?
Should already be fixed: https://github.com/primefaces/primefaces/issues/11806

**Comment by martin654** [comment]
Oh, I missed the ticket. It is probably already fixed in 14.0.1. I will wait till this minor version is released. Thanks
